Sources/FoundationEssentials/Calendar/Calendar.swift

```diff
@@ -483,7 +488,7 @@ public struct Calendar : Hashable, Equatable, Sendable {
     /// - returns: A new `DateInterval` if the starting time and duration of a component could be calculated, otherwise `nil`.
     @available(macOS 10.12, iOS 10.0, tvOS 10.0, watchOS 3.0, *)
     public func dateInterval(of component: Component, for date: Date) -> DateInterval? {
-        _calendar.dateInterval(of: component, for: date)
+        _calendar.dateInterval(of: component, for: date.capped)
```

Comment:
Is this the right place for this or in the specific calendar implementation?

Reply:
I do think so. I think the limits that `_calendar` can support is its implementation detail. The inner `_calendar` may be able to support beyond the capped date, but we still want to enforce the cap for `struct Calendar`'s clients for compatibility reasons here, regardless of what `_calendar` is